gulpfile.js

```diff
@@ -29,22 +34,125 @@ var tsCompileOptions = {
-    var process = child_process.exec(command + " " + args.join(" "));
+function spawnCommand(command, args, callback, silent, detached) {
+    var options = {};
+    if (detached) {
+        options.detached = true;
+        options.stdio = ["ignore"];
+    }
+    
+    var process = child_process.spawn(command, args, options);
+
+    process.stdout.on('data', function (data) {
+        if (!silent) console.log("" + data);
+    });
 
-    process.stdout.on('data', function(data) {
-        console.log("" + data);        
+    process.stderr.on('data', function (data) {
+        if (!silent) console.error("" + data);
     });
+
+    if (!detached) {
+        process.on('exit', function (code) {
+            callback && callback(code === 0 ? undefined : "Error code: " + code);
+        });
+    }
     
-    process.stderr.on('data', function(data) {
-        console.error("" + data);
+    return process;
+};
+
+function execCommand(command, args, callback, silent) {
+    var process = child_process.exec(command + " " + args.join(" "));
+
+    process.stdout.on('data', function (data) {
+        if (!silent) console.log("" + data);
     });
+
+    process.stderr.on('data', function (data) {
+        if (!silent) console.error("" + data);
+    });
+    
+    process.on('error', function (code) {
+        callback && callback(code);
+    })
     
-    process.on('exit', function(code) {
-        callback(code === 0 ? undefined : "Error code: " + code);
+    process.on('exit', function (code) {
+        callback && callback(code === 0 ? undefined : "Error code: " + code);
     });
+    
+    return process;
 };
 
+/**
+ * Executes a child process and returns its output in the promise as a string
+ */
+function execCommandWithPromise(command, options, logOutput) {
+    var deferred = Q.defer();
+
+    options = options || {};
+    options.maxBuffer = 1024 * 500;
+    // abort processes that run longer than five minutes
+    options.timeout = 5 * 60 * 1000;
+
+    console.log("Running command: " + command);
+    child_process.exec(command, options, (error, stdout, stderr) => {
+
+        if (logOutput) stdout && console.log(stdout);
+        stderr && console.error(stderr);
+
+        if (error) {
+            console.error(error);
+            deferred.reject(error);
+        } else {
+            deferred.resolve(stdout.toString());
+        }
+    });
+
+    return deferred.promise;
+}
+
+function getCommandLineFlag(optionName) {
+    for (var i = 0; i < process.argv.length; i++) {
+        if (process.argv[i].indexOf(optionName) === 0) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function getCommandLineOption(optionName, defaultValue) {
+    for (var i = 0; i < process.argv.length; i++) {
+        if (process.argv[i].indexOf(optionName) === 0) {
+            if (i + 1 < process.argv.length) {
+                return process.argv[i + 1];
+            }
+            break;
+        }
+    }
+    return defaultValue;
+}
```

Comment:
they were used by the emulator code, which is now in an entirely different file lol